app/controllers/clients/history_controller.rb

```diff
@@ -168,6 +168,8 @@ def set_pdf_dates
     end
 
     def set_client
+      not_authorized! and return unless current_user.present?
```

Comment:
This may break the PDF that gets generated for verification of homelessness.  (See the skip before actions.)